### PR TITLE
Implement request_vc DI response

### LINKED ISSUE #1929: Implement synchronous DI response for `request_vc`
### Context

Follow-up of https://github.com/litentry/litentry-parachain/issues/1883.

Once https://github.com/litentry/litentry-parachain/pull/1920 gets merged, we should add the DI response for `request_vc` too - please note, some of the VC metadata still needs to go on-chain (like `vcHash`), the DI response should mainly contain the VC payload.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by Kailai-Wang**
### Context

resolves #1929 

In fact, it will emit both the parachain event and synchronous DI response - similar to other trustedCalls.

It uses `H256` in the callback trustedCall too - I have no idea why `[u8; 32]` was used.

### CODE REVIEW

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts [hunk 1]
```diff
@@ -109,13 +111,13 @@ export async function runExample(keyPairType: KeypairType) {
     assert.isTrue(res.status.asTrustedOperationStatus[0].isInSidechainBlock);
     assert.equal(u8aToHex(res.status.asTrustedOperationStatus[1]), getTopHash(parachainApi, setUserShieldingKeyCall));
 
-    let decodedRes = parachainApi.createType(
+    const res1 = parachainApi.createType(
```
Comment (kziemianek) [inline tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: maybe `setUserShieldingKeyRes` ?

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts [hunk 2]
```diff
@@ -154,10 +156,10 @@ export async function runExample(keyPairType: KeypairType) {
     console.log('linkIdentity call returned', res.toHuman());
     assert.isTrue(res.do_watch.isFalse);
     assert.isTrue(res.status.asTrustedOperationStatus[0].isInSidechainBlock);
-    decodedRes = parachainApi.createType('LinkIdentityResponse', res.value) as unknown as LinkIdentityResponse;
-    assert.equal(decodedRes.account.toHex(), u8aToHex(alice.addressRaw));
-    assert.equal(decodedRes.req_ext_hash.toHex(), hash);
-    aesOutput = parseAesOutput(parachainApi, decodedRes.id_graph.toHex());
+    const res2 = parachainApi.createType('LinkIdentityResponse', res.value) as unknown as LinkIdentityResponse;
```
Comment (kziemianek) [inline tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: `linkIdentityRes` ?

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts [hunk 3]
```diff
@@ -109,13 +111,13 @@ export async function runExample(keyPairType: KeypairType) {
...
 
-    let decodedRes = parachainApi.createType(
+    const res1 = parachainApi.createType(
         'SetUserShieldingKeyResponse',
         res.value
     ) as unknown as SetUserShieldingKeyResponse;
-    assert.equal(decodedRes.account.toHex(), u8aToHex(alice.addressRaw));
-    assert.equal(decodedRes.req_ext_hash.toHex(), hash);
-    let aesOutput = parseAesOutput(parachainApi, decodedRes.id_graph.toHex());
+    assert.equal(res1.account.toHex(), u8aToHex(alice.addressRaw));
```
Comment (kziemianek) [inline tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: if these assertions are commnon, we could have an `util` function to do that

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts [hunk 4]
```diff
@@ -317,6 +319,33 @@ export async function runExample(keyPairType: KeypairType) {
...
+        alice,
+        aliceSubject,
+        parachainApi.createType('Assertion', { A1: null }).toHex(),
+        hash
+    );
+    res = await sendRequestFromTrustedCall(wsp, parachainApi, mrenclave, key, requestVcCall);
+    console.log('requestVcCall call returned', res.toHuman());
+    assert.isTrue(res.do_watch.isFalse);
+    assert.isTrue(res.status.asTrustedOperationStatus[0].isInSidechainBlock);
+    const res3 = parachainApi.createType('RequestVCResponse', res.value) as unknown as RequestVCResponse;
```
Comment (kziemianek) [inline tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: `requestVcResp` ?

File: tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs [hunk 5]
```diff
@@ -56,3 +56,13 @@ pub(crate) struct ActivateIdentityResponse {
 pub(crate) struct SetIdentityNetworksResponse {
 	pub req_ext_hash: H256,
 }
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq)]
+pub(crate) struct RequestVCResponse {
+	pub account: AccountId,
```
Comment (kziemianek) [inline tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs:62]: Why do we return `account` and `assertion`?
Doesn't caller know these values?

I missed that with previous PR regarding responses.
Probably for the sake of consistency with II but maybe there is another requirement ?

Comment (kziemianek) [reply tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs:62]: I'm in favour of having public interface as lean as possible

Comment (BillyWooo) [reply tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs:62]: I think it's easier for caller to find out  for which call this response is.

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts (hunk 1 above)
Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: I was struggling with the naming - `setUserShieldingKeyRes` just seems too long but meanwhile I can't find a better name (and short) - I wish ts could change declared types...

**Comment by Kailai-Wang** [comment]
Btw why do you think it's a breaking change? @grumpygreenguy
I thought it was compatible with the current client

File: tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs (hunk 5 above)
Comment (Kailai-Wang) [reply tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs:62]: Yes to be consistent with the II call.

Actually I'm thinking of using these `*Response` structs in the II extrinsic interfaces too - and if the `account` and `assertion` need to be exposed would require another discussion including IDHub colleagues.

File: tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts (hunk 1 above)
Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/examples/direct-invocation/example.ts]: Nvm let me change that...

**Review by kziemianek — APPROVED** [review]

**Comment by arieljbirnbaum** [comment]
> Btw why do you think it's a breaking change? @grumpygreenguy I thought it was compatible with the current client

It changes the encoding/metadata; shouldn't we consider _all_ such changes to be breaking?

**Comment by Kailai-Wang** [comment]
> > Btw why do you think it's a breaking change? @grumpygreenguy I thought it was compatible with the current client
> 
> It changes the encoding/metadata; shouldn't we consider _all_ such changes to be breaking?

Hmmm.. it shouldn't. Can you name the files that are breaking?

If you mean the `*metadata.json` change - IIRC it doesn't seem to be deterministic..

File: tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs (hunk 5 above)
Comment (kziemianek) [reply tee-worker/app-libs/stf/src/trusted_call_rpc_response.rs:62]: > I think it's easier for caller to find out for which call this response is.

We got this correlation identifier called `req_ext_hash`.

**Pull request merged**